libraries/Microsoft.Bot.Builder.LanguageGeneration/Templates.cs

```diff
@@ -391,15 +394,53 @@ private void CheckErrors()
             }
         }
 
-        private bool GetStrictModeFromOptions(IList<string> options)
+        private EvaluationOptions GetCurrentEvalOptions()
+        {
+            var opt = GetEvaluationOptions(Options);
+            foreach (var templates in References) 
+            {
+                var childOpt = GetEvaluationOptions(templates.Options);
+                if (childOpt.StrictMode != null)
+                {
+                    opt.StrictMode = childOpt.StrictMode;
+                }
+
+                if (childOpt.NullSubstitution != null)
+                {
+                    opt.NullSubstitution = childOpt.NullSubstitution;
+                }
+
+                if (childOpt.LineBreakStyle != null)
+                {
+                    opt.LineBreakStyle = childOpt.LineBreakStyle;
+                }
```

Comment:
The semantics of having children with modes seems murky at best.  Either we should have errors if people specify options more than once or we should make it so that you always use the previous options.  This semantic is just weird.

Reply:
I rewrited the rule completely.